cli/src/description_util.rs

```diff
@@ -31,12 +43,138 @@ JJ: Lines starting with "JJ: " (like this one) will be removed.
         settings,
     )?;
 
-    // Normalize line ending, remove leading and trailing blank lines.
-    let description = description
-        .lines()
-        .filter(|line| !line.starts_with("JJ: "))
-        .join("\n");
-    Ok(text_util::complete_newline(description.trim_matches('\n')))
+    Ok(cleanup_description(&description))
+}
+
+#[derive(Debug)]
+pub struct EditMultipleDescriptionsResult {
+    /// The parsed, formatted descriptions.
+    pub descriptions: HashMap<CommitId, String>,
+    /// Commit IDs that were expected while parsing the edited messages, but
+    /// which were not found.
+    pub missing: Vec<String>,
+    /// Commit IDs that were found multiple times while parsing the edited
+    /// messages.
+    pub duplicates: Vec<String>,
+    /// Commit IDs that were found while parsing the edited messages, but which
+    /// were not originally being edited.
+    pub unexpected: Vec<String>,
+}
+
+/// Edits the descriptions of the given commits in a single editor session.
+pub fn edit_multiple_descriptions(
+    tx: &mut WorkspaceCommandTransaction,
+    repo: &ReadonlyRepo,
+    commits: &[&Commit],
+    settings: &UserSettings,
+) -> Result<EditMultipleDescriptionsResult, CommandError> {
+    let mut commits_map = HashMap::new();
+    let mut output_chunks = Vec::new();
+
+    for &commit in commits.iter() {
+        let commit_hash = short_commit_hash(commit.id());
+        if commits.len() > 1 {
+            output_chunks.push(format!("JJ: describe {} -------\n", commit_hash.clone()));
+        }
+        commits_map.insert(commit_hash, commit.id());
+        let mut commit_builder = tx.mut_repo().rewrite_commit(settings, commit).detach();
+        if commit_builder.description().is_empty() {
+            commit_builder.set_description(settings.default_description());
+        }
+        let temp_commit = commit_builder.write_hidden()?;
```

Comment:
nit: maybe better to create `temp_commit` by caller and pass `[(old_commit_id, temp_commit)]` to this function.

It's not super important, but `if reset_author`, the rendered description template should include the new author.